datafusion/datasource-parquet/src/file_format.rs

```diff
@@ -1749,16 +1732,16 @@ fn spawn_parquet_parallel_serialization_task(
                     // Do not surface error from closed channel (means something
                     // else hit an error, and the plan is shutting down).
```

Comment:
Line 1726 currently seems to cause a deadlock the second time it's hit in `write_parquet_with_small_rg_size` test. I imagine a thread won't release column writers while another is trying to use them looping forever or some version of that. I'm still trying to understand what exactly happens.